File: benchmarks/pandas/bench_read_csv.py
"""Benchmark: read_csv — parse a 100k-row CSV file"""
import json, time, os, tempfile
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 2
ITERATIONS = 5

# Build CSV file
tmp_path = "/tmp/gh-aw/agent/bench_read_csv.csv"
with open(tmp_path, "w") as f:
    f.write("id,value,label\n")
    for i in range(ROWS):
        f.write(f"{i},{i * 1.1:.4f},cat_{i % 50}\n")

for _ in range(WARMUP):
    pd.read_csv(tmp_path)

start = time.perf_counter()
for _ in range(ITERATIONS):
    pd.read_csv(tmp_path)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "read_csv",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_read_csv.ts
/**
 * Benchmark: read_csv — parse a 100k-row CSV string
 */
import { readCsv } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 2;
const ITERATIONS = 5;

// Build CSV string
const lines = ["id,value,label"];
for (let i = 0; i < ROWS; i++) {
  lines.push(`${i},${(i * 1.1).toFixed(4)},cat_${i % 50}`);
}
const csvContent = lines.join("\n");

for (let i = 0; i < WARMUP; i++) {
  readCsv(csvContent);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  readCsv(csvContent);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "read_csv",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
